modules.d/01fips/fips.sh

```diff
@@ -81,6 +94,33 @@ do_rhevh_check() {
     return 0
 }
 
+do_uki_check() {
+    KVER=$(uname -r)
+    # UKI are placed in $ESP\EFI\Linux\<intall-tag>-<uname-r>.efi
+    if ! [ "$FIPS_MOUNTED_BOOT" = 1 ]; then
+        warn "Failed to mount ESP for doing UKI integrity check"
+        return 1
+    fi
+
+    uki_checked=0
+    for UKIpath in /boot/EFI/Linux/*-"$KVER".efi; do
+        [ -f "$UKIpath" ] || break
+
+        UKI=${UKIpath##*/}
+        UKIHMAC=."$UKI".hmac
+
+        fips_info "checking $UKIHMAC"
+        (cd /boot/EFI/Linux/ && sha512hmac -c "$UKIHMAC") || return 1
```

Comment:
Please note that this will end on first unsuccesful check. Is that intentional? Are multiple UKIpaths (un)expected?

If we just need to check one, we could simply `return 0` here instead, and `return 1` bellow and drop the `uki_checked` variable.

Reply:
This is intentional. UKIs are installed to $ESP/EFI/Linux/<machine-id>-<uname-r>.efi but e.g. when machine id changes (and the image is moved to a different host) + kernel is reinstalled (sounds already very unlikely, but still) we may get two UKIs of the same version. As the kernel is already booted, we can't say for sure which one was it, especially when a bootloader with menu (e.g. sb-boot) was used. So for FIPS integrity check, I decided that we can be on the safe side and check **all** UKIs of the same version.

Reply:
Just to clear thing up (probably deserves a comment), you want all of them pass? (That's what it currently does.)

Reply:
Yes, this is the intention: to be on the safe side, let's make them ALL pass the check. I've added a comment explaining the decision.